### PR TITLE
[CC-27909] fetch: Add implicit auth for gcp

### PR DESCRIPTION
**Pull request by Jeremyyang920**
This commit allows the gcp resource to create an import url for implicit auth. This will be used when users do not specify the gcp env var or have a credential file in the well known location.

Release Note: Adds implicit auth for gcp

### CODE REVIEW

File: fetch/datablobstorage/gcp.go [hunk 1]
```diff
@@ -197,6 +197,19 @@ type gcpResource struct {
 }
 
 func (r *gcpResource) ImportURL() (string, error) {
+	// If we do not have the GOOGLE_APPLICATION_CREDENTIALS env var
```
Comment (ryanluu12345) [inline fetch/datablobstorage/gcp.go:200]: Can be in another PR, but worth adding an e2e test with this for the GCP case

Comment (ryanluu12345) [reply fetch/datablobstorage/gcp.go:200]: Tracking here: https://docs.google.com/spreadsheets/d/1YsLWaIYgQReBWzxid6MKrsRYoTDeEjRWDGFNgAQl0y8/edit?usp=sharing

**Review by ryanluu12345 — APPROVED** [review]

File: fetch/datablobstorage/gcp.go (hunk 1 above)
Comment (Jeremyyang920) [reply fetch/datablobstorage/gcp.go:200]: You can't really add an e2e for this. The implicit auth either happens on a GCE worker directly, or you are exporting the json service account file in an env var. Unless you are talking about the env var case. But in which case we then have to check in credentials to the repo which we wont be able to do. This tests more the GCP SDK more than it does the actual fetch code.

**Review by ZhouXing19 — APPROVED** [review]
